OSDOCS-9428: install multus running cluster MicroShift (#74739)
openshift/openshift-docs: modules/microshift-install-multus-running-cluster.adoc

// Module included in the following assemblies:
//
// * microshift_networking/microshift-cni-multus.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-multus-installing-on-running-cluster_{context}"]
= Installing the Multus CNI plugin on an running cluster

If you want to attach additional networks to a pod for high-performance network configurations, you can install the {microshift-short} Multus RPM package. After installation, a host restart is required to recreate all the pods with the Multus annotation.

[IMPORTANT]
====

> The Multus RPM cannot be removed after you install it.

Reviewer: ```suggestion
The Multus RPM should not be removed after you install it.
```

====

.Procedure

. Install the Multus RPM package by running the following command:
+
[source,terminal]
----
$ sudo dnf install microshift-multus
----
+
[TIP]
====
If you create your custom resources (CRs) before completing your installation of Multus, you can avoid restarting the service a second time to apply them.
====

. To apply the manifest from the package to an active cluster, restart the host by running the following command:
+
[source,terminal]
----
$ sudo systemctl restart
----

.Verification

> . Ensure that the restarted pods have the Multus annotation by running the following command:

Reviewer: Do we need to say that we just expect non-empty output?

// Module included in the following assemblies:
//
// * microshift_networking/microshift-cni-multus.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-multus-installing-on-running-cluster_{context}"]
= Installing the Multus CNI plugin on an running cluster

If you want to attach additional networks to a pod for high-performance network configurations, you can install the {microshift-short} Multus RPM package. After installation, a host restart is required to recreate all the pods with the Multus annotation.

[IMPORTANT]
====
The Multus RPM should not be removed after you install it.
====

.Procedure

. Install the Multus RPM package by running the following command:
+
[source,terminal]
----
$ sudo dnf install microshift-multus
----
+
[TIP]
====
If you create your custom resources (CRs) before completing your installation of Multus, you can avoid restarting the service a second time to apply them.
====

. To apply the manifest from the package to an active cluster, restart the host by running the following command:
+
[source,terminal]
----
$ sudo systemctl restart
----

.Verification

> In general, any non-empty output is expected with this step.

Reviewer 2: The verification steps and output is not clear for the user. Should we say both multus pod and multus dhcp-daemon will be created after systemctl restarting. And include below output
```
$ oc get pod -A | grep multus
openshift-multus                       dhcp-daemon-ktzqf                         1/1     Running   0          45h
openshift-multus                       multus-4frf4                              1/1     Running   0          45h

```
@pmtk WDYT?

Author: That looks better to me from a user perspective. We're just checking the packages at this point, not any config.

Author: @pmtk I'll put this PR in peer review; if you aren't happy with this update, LMK.

:_mod-docs-content-type: CONCEPT
[id="microshift-multus-installing-on-running-cluster_{context}"]
= Installing the Multus CNI plugin on an running cluster

If you want to attach additional networks to a pod for high-performance network configurations, you can install the{microshift-short} Multus RPM package. After installation, a host restart is required to recreate all the pods with the Multus annotation.

[IMPORTANT]
====
The Multus RPM cannot be removed after you install it.
====

.Procedure

. Install the Multus RPM package by running the following command:
+
[source,terminal]
----
$ sudo dnf install microshift-multus
----
+
[TIP]
====
If you create your custom resource definition (CRD) and custom resources (CRs) before completing your installation of Multus, you can avoid restarting the service a second time to apply them.
====

. To apply the manifest from the package to an active cluster, restart the host by running the following command:
+
[source,terminal]
----
$ sudo systemctl restart
----

.Verification

. Ensure that the restarted pods have the Multus annotation by running the following command:
+
[source,terminal]
----

> $ oc get pods -a

Reviewer: `-a` doesn't exist, but we could use:
```
oc get pods -A -o=jsonpath='{ .items[*].metadata.annotations.k8s\.v1\.cni\.cncf\.io/network-status }'
```
output with multus:
```
[{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.16",
        "fd01:0:0:1::10"
    ],
    "mac": "0a:58:0a:2a:00:10",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.7"
    ],
    "mac": "0a:58:0a:2a:00:07",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.9"
    ],
    "mac": "0a:58:0a:2a:00:09",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.10"
    ],
    "mac": "0a:58:0a:2a:00:0a",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.3"
    ],
    "mac": "0a:58:0a:2a:00:03",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.4"
    ],
    "mac": "0a:58:0a:2a:00:04",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.5"
    ],
    "mac": "0a:58:0a:2a:00:05",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.8"
    ],
    "mac": "0a:58:0a:2a:00:08",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.6"
    ],
    "mac": "0a:58:0a:2a:00:06",
    "default": true,
    "dns": {}
}] [{
    "name": "ovn-kubernetes",
    "interface": "eth0",
    "ips": [
        "10.42.0.11",
        "fd01:0:0:1::b"
    ],
    "mac": "0a:58:0a:2a:00:0b",
    "default": true,
    "dns": {}
}]⏎                                  
```
without multus it'd be empty

// Module included in the following assemblies:
//
// * microshift_networking/microshift-cni-multus.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-multus-installing-on-running-cluster_{context}"]

> = Installing the Multus CNI plugin on an running cluster

Reviewer 3: ```suggestion
= Installing the Multus CNI plugin on a running cluster
```

If you want to attach additional networks to a pod for high-performance network configurations, you can install the {microshift-short} Multus RPM package. After installation, a host restart is required to recreate all the pods with the Multus annotation.

[IMPORTANT]
====

> The Multus RPM should not be removed after you install it.

Reviewer 3: ```suggestion
The Multus RPM should not be removed after installation.
```

====

.Procedure

. Install the Multus RPM package by running the following command:
+
[source,terminal]
----

> $ sudo dnf install microshift-multus

Reviewer 4: If the customer needs `sudo`/`root` permissions, it is a good idea to call that out as prerequisite. This is inconsistently applied in reviews, but I think it is a good practice. That said, I know that there are some exceptions with MicroShift, so feel free to disregard if this comment is not relevant.

Author: That's a good callout. There are no `roles` in MicroShift, but the admin still needs root access to the host in this case. Shall add.

----
+
[TIP]
====

> If you create your custom resources (CRs) before completing your installation of Multus, you can avoid restarting the service a second time to apply them.

Reviewer 3: I'm not sure which of these versions best fits the process, but one of them might be a more streamlined version that you could use.
```suggestion
Create your custom resources (CRs) before installing Multus to avoid restarting the service a second time to apply them.
```
```suggestion
Create your custom resources (CRs) before Multus installs completely to avoid restarting the service a second time to apply them.
```
```suggestion
Create your custom resources (CRs) before while installing Multus to avoid restarting the service a second time to apply them.
```

Author: Hmm, none of those quite fit because it is literally one commmand to install the RPM. BUT restarting the host is quite disruptive and that's what we're trying to get at here. Went with:

If you create your custom resources (CRs) for additional networks now, you can complete your installation and apply configurations with one restart.

====

> . To apply the manifest from the package to an active cluster, restart the host by running the following command:

Reviewer 3: ```suggestion
. Apply the package manifest to an active cluster and restart the host by running the following command:
```
Does this change the meaning to much? If so, maybe we could try:
```suggestion
. To apply the package manifest to an active cluster, restart the host by running the following command:
```

Author: Second one is accurate.

+
[source,terminal]
----
$ sudo systemctl restart
----

.Verification

> . Ensure that the Multus CNI plugin components are created after restarting by running the following command:

Reviewer 3: ```suggestion
. After restarting, ensure that the Multus CNI plugin components are created by running the following command:
```
Just a thought to avoid all the gerunds at the end of the sentence.

// Module included in the following assemblies:
//
// * microshift_networking/microshift-cni-multus.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-multus-installing-on-running-cluster_{context}"]
= Installing the Multus CNI plugin on an running cluster

If you want to attach additional networks to a pod for high-performance network configurations, you can install the {microshift-short} Multus RPM package. After installation, a host restart is required to recreate all the pods with the Multus annotation.

[IMPORTANT]
====
The Multus RPM should not be removed after you install it.
====

.Procedure

. Install the Multus RPM package by running the following command:
+
[source,terminal]
----
$ sudo dnf install microshift-multus
----
+
[TIP]
====
If you create your custom resources (CRs) before completing your installation of Multus, you can avoid restarting the service a second time to apply them.
====

. To apply the manifest from the package to an active cluster, restart the host by running the following command:
+
[source,terminal]
----
$ sudo systemctl restart
----

.Verification

> In general, any non-empty output is expected with this step.

Reviewer 2: Do we still need this statement? `In general, any non-empty output is expected with this step.`

Author: No. Good catch.
